api/src/main/java/org/ehcache/spi/service/MaintainableService.java

```diff
@@ -22,11 +22,20 @@
  */
 @PluralService
 public interface MaintainableService extends Service {
+
+  enum MaintenanceScope {
+    /** Will impact the cache manager */
+    CACHE_MANAGER,
+    /** Will impact one or many caches */
+    CACHE
+  }
+
   /**
    * Start this service for maintenance, based on its default configuration.
-   *
    * @param serviceProvider enables to depend on other maintainable services
+   * @param maintenanceScope the scope of the maintenance
+   *
    */
-  void startForMaintenance(ServiceProvider<MaintainableService> serviceProvider);
+  void startForMaintenance(ServiceProvider<MaintainableService> serviceProvider, MaintenanceScope maintenanceScope);
```

Comment:
I think the generic type here should be `ServiceProvider<? super MaintainableService>`, since it's fine to pass in a more generic provider here, just not okay for the consumer to use it as a wider type.  Also, isn't this the value that there is in the generic type on ServiceProvider?